src/catalog/src/durable/persist.rs

```diff
@@ -694,6 +716,9 @@ impl ApplyUpdate<StateUpdateKindJson> for UnopenedCatalogStateInner {
         current_epoch: &mut FenceableEpoch,
         _metrics: &Arc<Metrics>,
     ) -> Result<Option<StateUpdate<StateUpdateKindJson>>, DurableCatalogError> {
+        // TODO(jkosh44) It's a bit unfortunate that we have to clone all updates to attempt to
+        // convert them into a `StateUpdateKind` and cache a very small subset of them. It would
+        // be better if we could figure out a way not to clone everything.
```

Comment:
I'm thinking that we can probably inspect the JSON itself to see what the label is and only attempt to convert it based on the label.